packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -1,42 +1,66 @@
-import React, { ReactElement, useState } from "react";
+import React, { ReactElement, useRef, useState } from "react";
 import classnames from "classnames";
+import ReactDOM from "react-dom";
+import { usePopper } from "react-popper";
+import { useOnKeyDown } from "@jobber/hooks/useOnKeyDown";
 import styles from "./ComboboxContent.css";
 import { Icon } from "../../../Icon";
 import { ComboboxContext } from "../../ComboboxProvider";
+import { ComboboxOption } from "../../Combobox.types";
 
 interface ComboboxContentProps {
   /**
-   * options for the combobox
+   * List of selectable options to display.
    */
-  readonly options: string[];
+  readonly options: ComboboxOption[];
 
   /**
-   * onSelection handler.
+   * Callback function invoked upon the selection of an option. Provides the selected option as an argument.
    */
-  readonly onSelection: (selection: string) => void;
+  readonly onSelection: (selection: ComboboxOption) => void;
 
   /**
-   * actions
+   * Optional action button(s) to display at the bottom of the list.
    */
-  readonly children: ReactElement | ReactElement[];
+  readonly children?: ReactElement | ReactElement[];
+
+  /**
+   * Placeholder text to display in the search input. Defaults to "Search".
+   */
+  readonly searchPlaceholder?: string;
 }
 
 export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
-  const { open } = React.useContext(ComboboxContext);
+  const {
+    searchValue,
+    setSearchValue,
+    selectedOption,
+    setSelectedOption,
+    open,
+    setOpen,
+    popperRef,
+    popperStyles,
+    attributes,
+  } = useComboboxContent();
 
   const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
+    option.label.toLowerCase().includes(searchValue.toLowerCase()),
   );
 
-  return (
-    <div className={classnames(styles.content, !open && styles.hidden)}>
+  const template = (
+    <div
+      ref={popperRef}
+      data-testid="combobox-content"
+      className={classnames(styles.content, !open && styles.hidden)}
```

Comment:
I tried doing something higher above this component that would prevent it from rendering at all in `Combobox`

```
    <ComboboxContextProvider>
      {triggerElement}
      <ComboboxContentWrapper>
         {contentElement}
      </ComboboxContentWrapper/>
    </ComboboxContextProvider>

...
const ComboboxContentWrapper = (props: {children: ReactNode}): JSX.Element {
  const { open } = React.useContext(ComboboxContext);

  return (
    <>
     {open && (
       {props.children}
     )}
   </>
  )
}
```

so that way yeah it just doesn't render at all. we have to have a component wrapping it because since this is where the provider is first provided, it can't be used in `Combobox` yet

the problem here is that popper seems to get super lost and fails to adjust the location of the popper element which results in it being overtop over the trigger since it's `absolute top: 0 left: 0` and that's relative to the first parent, not relative to the container due to the `ReactDOM` placing it out there